pandas/core/arrays/base.py

```diff
@@ -430,9 +443,10 @@ def __ne__(self, other: Any) -> ArrayLike:  # type: ignore[override]
 
     def to_numpy(
         self,
-        dtype: Optional[Dtype] = None,
+        dtype: Optional[NpDtype] = None,
         copy: bool = False,
-        na_value=lib.no_default,
+        na_value: Optional[Any] = lib.no_default,
+        **kwargs: Any,
```

Comment:
So I put that in there because of `pandas/tests/extension/decimal/array.py` which adds an extra argument  to `to_numpy()` .  I then recoded `to_numpy()` there to look at the `**kwargs`  rather than use the explicit argument.  

I also removed the test.

Reply:
this is to remove the mypy error in Index.to_numpy in pandas/core/base.py? if a user passes a kwarg there that is not accepted by the underlying EA, don't we want it to raise?

Reply:
Shouldn't it be the 

> this is to remove the mypy error in Index.to_numpy in pandas/core/base.py? if a user passes a kwarg there that is not accepted by the underlying EA, don't we want it to raise?

We could check in `pandas/core/arrays/base.py` in `to_numpy()` and if `kwargs` are present, we raise.  Is that what you are suggesting?

Reply:
not really. Python will raise a TypeError if addional kwargs are passed. we don't need to add code to do that.

```
>>> def my_func(foo=None):
...     pass
...
>>> my_func(bar = 5)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: my_func() got an unexpected keyword argument 'bar'
>>>
```

mypy is a static type checker that helps prevent TypeError

so imo we leave it as it is and ignore mypy.

Reply:
OK, I will remove the `**kwargs` from `to_numpy`